### PR TITLE
envoy: init at 1.16.2

### PR DESCRIPTION
**Pull request by lukegb**
###### Motivation for this change

Add [Envoy](https://github.com/envoyproxy/envoy), a cloud-native proxy.

I'm adding this because I want to add [Pomerium](https://www.pomerium.io/), and Envoy is a dependency (it's the data-plane proxy that Pomerium uses).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lukegb** [comment]
@ofborg build envoy

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108741` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
13 packages marked as broken and skipped:
</summary>
<ul>
<li>bazel-watcher</li>
<li>python37Packages.baselines</li>
<li>python37Packages.graph_nets</li>
<li>python37Packages.mask-rcnn</li>
<li>python37Packages.optuna</li>
<li>python37Packages.rl-coach</li>
<li>python38Packages.baselines</li>
<li>python38Packages.graph_nets</li>
<li>python38Packages.mask-rcnn</li>
<li>python38Packages.optuna</li>
<li>python38Packages.rl-coach</li>
<li>python39Packages.baselines</li>
<li>python39Packages.graph_nets</li>
</ul>
</details>
<details>

  <summary>29 packages failed to build and are new build failures:</summary>
  <ul>
<li>bazel-remote: log <a href="https://termbin.com/kzun">https://termbin.com/kzun</a></li>
<li>python37Packages.arviz: log was empty</li>
<li>python37Packages.edward: log was empty</li>
<li>python37Packages.pymc3: log was empty</li>
<li>python37Packages.tensorflow: log was empty</li>
<li>(python37Packages.tensorflow-build: log was empty</li>
<li>,python37Packages.tensorflow-build_2: log was empty</li>
<li>,python37Packages.tensorflowWithoutCuda: log was empty</li>
<li>,python37Packages.tensorflow_2): log was empty</li>
<li>python37Packages.tensorflow-probability: log was empty</li>
<li>python37Packages.tflearn: log was empty</li>
<li>python38Packages.arviz: log was empty</li>
<li>python38Packages.edward: log was empty</li>
<li>python38Packages.pymc3: log was empty</li>
<li>python38Packages.tensorflow: log was empty</li>
<li>(python38Packages.tensorflow-build: log was empty</li>
<li>,python38Packages.tensorflow-build_2: log was empty</li>
<li>,python38Packages.tensorflowWithoutCuda: log was empty</li>
<li>,python38Packages.tensorflow_2): log was empty</li>
<li>python38Packages.tensorflow-probability: log was empty</li>
<li>python38Packages.tflearn: log was empty</li>
<li>python39Packages.edward: log was empty</li>
<li>python39Packages.tensorflow: log was empty</li>
<li>(python39Packages.tensorflow-build: log was empty</li>
<li>,python39Packages.tensorflow-build_2: log was empty</li>
<li>,python39Packages.tensorflowWithoutCuda: log was empty</li>
<li>,python39Packages.tensorflow_2): log was empty</li>
<li>python39Packages.tensorflow-probability: log was empty</li>
<li>python39Packages.tflearn: log was empty</li>
  </ul>
</details>

**Comment by lukegb** [comment]
@ofborg build envoy

**Comment by lukegb** [comment]
@ofborg build envoy

**Comment by lukegb** [comment]
@ofborg eval

**Comment by lukegb** [comment]
Result of `nixpkgs-review pr 108741` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages marked as broken and skipped:</summary>
  <ul>
    <li>bazel-watcher</li>
    <li>python37Packages.baselines</li>
    <li>python37Packages.graph_nets</li>
    <li>python37Packages.mask-rcnn</li>
    <li>python37Packages.optuna</li>
    <li>python37Packages.rl-coach</li>
    <li>python38Packages.baselines</li>
    <li>python38Packages.graph_nets</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.rl-coach</li>
    <li>python39Packages.baselines</li>
    <li>python39Packages.graph_nets</li>
  </ul>
</details>
<details>
  <summary>18 packages failed to build:</summary>
  <ul>
    <li>bazel-remote</li>
    <li>python37Packages.arviz</li>
    <li>python37Packages.edward</li>
    <li>python37Packages.pymc3</li>
    <li>python37Packages.tensorflow (python37Packages.tensorflow-build ,python37Packages.tensorflow-build_2 ,python37Packages.tensorflowWithoutCuda ,python37Packages.tensorflow_2)</li>
    <li>python37Packages.tensorflow-probability</li>
    <li>python37Packages.tflearn</li>
    <li>python38Packages.arviz</li>
    <li>python38Packages.edward</li>
    <li>python38Packages.pymc3</li>
    <li>python38Packages.tensorflow (python38Packages.tensorflow-build ,python38Packages.tensorflow-build_2 ,python38Packages.tensorflowWithoutCuda ,python38Packages.tensorflow_2)</li>
    <li>python38Packages.tensorflow-probability</li>
    <li>python38Packages.tflearn</li>
    <li>python39Packages.edward</li>
    <li>python39Packages.tensorflow (python39Packages.tensorflow-build ,python39Packages.tensorflow-build_2 ,python39Packages.tensorflowWithoutCuda ,python39Packages.tensorflow_2)</li>
    <li>python39Packages.tensorflow-probability</li>
    <li>python39Packages.tflearn</li>
    <li>tts</li>
  </ul>
</details>

File: pkgs/servers/http/envoy/default.nix [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    # These are contained in a attrset of their own to make it obvious that
+    # people should update both.
+    version = "1.16.2";
+    commit = "e98e41a8e168af7acae8079fc0cd68155f699aa3";
+  };
+in
+buildBazelPackage rec {
+  pname = "envoy";
+  version = srcVer.version;
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/servers/http/envoy/default.nix:23]:
```suggestion
  version = srcVer.version;

  src = fetchFromGitHub {
```

**Comment by lukegb** [comment]
Result of `nixpkgs-review pr 108741` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python37Packages.rl-coach</li>
    <li>python38Packages.rl-coach</li>
  </ul>
</details>
<details>
  <summary>28 packages failed to build:</summary>
  <ul>
    <li>python37Packages.arviz</li>
    <li>python37Packages.baselines</li>
    <li>python37Packages.dm-sonnet</li>
    <li>python37Packages.edward</li>
    <li>python37Packages.graph_nets</li>
    <li>python37Packages.mask-rcnn</li>
    <li>python37Packages.optuna</li>
    <li>python37Packages.pymc3</li>
    <li>python37Packages.tensorflow (python37Packages.tensorflow-build ,python37Packages.tensorflow-build_2 ,python37Packages.tensorflowWithoutCuda ,python37Packages.tensorflow_2)</li>
    <li>python37Packages.tensorflow-probability</li>
    <li>python37Packages.tensorflowWithCuda</li>
    <li>python37Packages.tflearn</li>
    <li>python38Packages.arviz</li>
    <li>python38Packages.baselines</li>
    <li>python38Packages.dm-sonnet</li>
    <li>python38Packages.edward</li>
    <li>python38Packages.graph_nets</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.pymc3</li>
    <li>python38Packages.tensorflow (python38Packages.tensorflow-build ,python38Packages.tensorflow-build_2 ,python38Packages.tensorflowWithoutCuda ,python38Packages.tensorflow_2)</li>
    <li>python38Packages.tensorflow-probability</li>
    <li>python38Packages.tflearn</li>
    <li>python39Packages.dm-sonnet</li>
    <li>python39Packages.graph_nets</li>
    <li>python39Packages.tensorflow-probability</li>
    <li>python39Packages.tensorflowWithCuda</li>
    <li>tts</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>bazel-remote</li>
    <li>bazel-watcher</li>
    <li>envoy</li>
    <li>gvisor</li>
    <li>python38Packages.tensorflowWithCuda</li>
    <li>python39Packages.baselines</li>
    <li>python39Packages.edward</li>
    <li>python39Packages.tensorflow (python39Packages.tensorflow-build ,python39Packages.tensorflow-build_2 ,python39Packages.tensorflowWithoutCuda ,python39Packages.tensorflow_2)</li>
    <li>python39Packages.tflearn</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108741` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.rl-coach</li>
<li>python38Packages.rl-coach</li>
</ul>
</details>
<details>

  <summary>45 packages failed to build and are new build failures:</summary>
  <ul>
<li>envoy: log was empty</li>
<li>python37Packages.arviz: log was empty</li>
<li>python37Packages.baselines: log was empty</li>
<li>python37Packages.dm-sonnet: log was empty</li>
<li>python37Packages.edward: log was empty</li>
<li>python37Packages.graph_nets: log was empty</li>
<li>python37Packages.mask-rcnn: log was empty</li>
<li>python37Packages.optuna: log was empty</li>
<li>python37Packages.pymc3: log was empty</li>
<li>python37Packages.tensorflow: log was empty</li>
<li>(python37Packages.tensorflow-build: log was empty</li>
<li>,python37Packages.tensorflow-build_2: log was empty</li>
<li>,python37Packages.tensorflowWithoutCuda: log was empty</li>
<li>,python37Packages.tensorflow_2): log was empty</li>
<li>python37Packages.tensorflow-probability: log was empty</li>
<li>python37Packages.tensorflowWithCuda: log was empty</li>
<li>python37Packages.tflearn: log was empty</li>
<li>python38Packages.arviz: log was empty</li>
<li>python38Packages.baselines: log was empty</li>
<li>python38Packages.dm-sonnet: log was empty</li>
<li>python38Packages.edward: log was empty</li>
<li>python38Packages.graph_nets: log was empty</li>
<li>python38Packages.mask-rcnn: log was empty</li>
<li>python38Packages.optuna: log was empty</li>
<li>python38Packages.pymc3: log was empty</li>
<li>python38Packages.tensorflow: log was empty</li>
<li>(python38Packages.tensorflow-build: log was empty</li>
<li>,python38Packages.tensorflow-build_2: log was empty</li>
<li>,python38Packages.tensorflowWithoutCuda: log was empty</li>
<li>,python38Packages.tensorflow_2): log was empty</li>
<li>python38Packages.tensorflow-probability: log was empty</li>
<li>python38Packages.tensorflowWithCuda: log was empty</li>
<li>python38Packages.tflearn: log was empty</li>
<li>python39Packages.baselines: log was empty</li>
<li>python39Packages.dm-sonnet: log was empty</li>
<li>python39Packages.edward: log was empty</li>
<li>python39Packages.graph_nets: log was empty</li>
<li>python39Packages.tensorflow: log was empty</li>
<li>(python39Packages.tensorflow-build: log was empty</li>
<li>,python39Packages.tensorflow-build_2: log was empty</li>
<li>,python39Packages.tensorflowWithoutCuda: log was empty</li>
<li>,python39Packages.tensorflow_2): log was empty</li>
<li>python39Packages.tensorflow-probability: log was empty</li>
<li>python39Packages.tensorflowWithCuda: log was empty</li>
<li>python39Packages.tflearn: log was empty</li>
  </ul>
</details>
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>bazel-remote</li>
<li>bazel-watcher</li>
<li>gvisor</li>
</ul>
</details>

**Comment by lukegb** [comment]
Result of `nixpkgs-review pr 108741` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python37Packages.rl-coach</li>
    <li>python38Packages.rl-coach</li>
  </ul>
</details>
<details>
  <summary>28 packages failed to build:</summary>
  <ul>
    <li>python37Packages.arviz</li>
    <li>python37Packages.baselines</li>
    <li>python37Packages.dm-sonnet</li>
    <li>python37Packages.edward</li>
    <li>python37Packages.graph_nets</li>
    <li>python37Packages.mask-rcnn</li>
    <li>python37Packages.optuna</li>
    <li>python37Packages.pymc3</li>
    <li>python37Packages.tensorflow (python37Packages.tensorflow-build ,python37Packages.tensorflow-build_2 ,python37Packages.tensorflowWithoutCuda ,python37Packages.tensorflow_2)</li>
    <li>python37Packages.tensorflow-probability</li>
    <li>python37Packages.tensorflowWithCuda</li>
    <li>python37Packages.tflearn</li>
    <li>python38Packages.arviz</li>
    <li>python38Packages.baselines</li>
    <li>python38Packages.dm-sonnet</li>
    <li>python38Packages.edward</li>
    <li>python38Packages.graph_nets</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.pymc3</li>
    <li>python38Packages.tensorflow (python38Packages.tensorflow-build ,python38Packages.tensorflow-build_2 ,python38Packages.tensorflowWithoutCuda ,python38Packages.tensorflow_2)</li>
    <li>python38Packages.tensorflow-probability</li>
    <li>python38Packages.tflearn</li>
    <li>python39Packages.dm-sonnet</li>
    <li>python39Packages.graph_nets</li>
    <li>python39Packages.tensorflow-probability</li>
    <li>python39Packages.tensorflowWithCuda</li>
    <li>tts</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>bazel-remote</li>
    <li>bazel-watcher</li>
    <li>envoy</li>
    <li>gvisor</li>
    <li>python38Packages.tensorflowWithCuda</li>
    <li>python39Packages.baselines</li>
    <li>python39Packages.edward</li>
    <li>python39Packages.tensorflow (python39Packages.tensorflow-build ,python39Packages.tensorflow-build_2 ,python39Packages.tensorflowWithoutCuda ,python39Packages.tensorflow_2)</li>
    <li>python39Packages.tflearn</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
```
hash mismatch in fixed-output derivation '/nix/store/cgxm8fc7fyx20n0f8j6jbjiibmyzs7nk-envoy-1.16.2-deps.tar.gz':
  wanted: sha256:0k2npf8f68iladnrai8jnfqc76440lpvc5sw1qv8bjpcrg97i4n7
  got:    sha256:1ycxrgj8nzv9j4x6g1clgg99id6hbq0399jkz1bs0673v60mjf95
cannot build derivation '/nix/store/y0yp6lmhqqbi5bkmqxc7v6rqf6sg0j80-envoy-1.16.2.drv': 1 dependencies couldn't be built
```

**Comment by lukegb** [comment]
@ofborg build envoy

**Comment by lukegb** [comment]
Hooray. I have now managed to remove all the outstanding references into the nix store from the envoy.deps build, which should avoid the hash changing as dependency versions are bumped.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108741` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
27 packages marked as broken and skipped:
</summary>
<ul>
<li>bazel-watcher</li>
<li>python37Packages.arviz</li>
<li>python37Packages.baselines</li>
<li>python37Packages.graph_nets</li>
<li>python37Packages.mask-rcnn</li>
<li>python37Packages.optuna</li>
<li>python37Packages.pymc3</li>
<li>python37Packages.rl-coach</li>
<li>python37Packages.tensorflow-probability</li>
<li>python38Packages.arviz</li>
<li>python38Packages.baselines</li>
<li>python38Packages.graph_nets</li>
<li>python38Packages.mask-rcnn</li>
<li>python38Packages.optuna</li>
<li>python38Packages.pymc3</li>
<li>python38Packages.rl-coach</li>
<li>python38Packages.tensorflow-probability</li>
<li>python39Packages.baselines</li>
<li>python39Packages.edward</li>
<li>python39Packages.graph_nets</li>
<li>python39Packages.tensorflow</li>
<li>python39Packages.tensorflow-build</li>
<li>python39Packages.tensorflow-build_2</li>
<li>python39Packages.tensorflow-probability</li>
<li>python39Packages.tensorflowWithoutCuda</li>
<li>python39Packages.tensorflow_2</li>
<li>python39Packages.tflearn</li>
</ul>
</details>
<details>

  <summary>15 packages failed to build and are new build failures:</summary>
  <ul>
<li>bazel-remote: log <a href="https://termbin.com/57hm">https://termbin.com/57hm</a></li>
<li>python37Packages.edward: log was empty</li>
<li>python37Packages.tensorflow: log was empty</li>
<li>(python37Packages.tensorflow-build: log was empty</li>
<li>,python37Packages.tensorflow-build_2: log was empty</li>
<li>,python37Packages.tensorflowWithoutCuda: log was empty</li>
<li>,python37Packages.tensorflow_2): log was empty</li>
<li>python37Packages.tflearn: log was empty</li>
<li>python38Packages.edward: log was empty</li>
<li>python38Packages.tensorflow: log was empty</li>
<li>(python38Packages.tensorflow-build: log was empty</li>
<li>,python38Packages.tensorflow-build_2: log was empty</li>
<li>,python38Packages.tensorflowWithoutCuda: log was empty</li>
<li>,python38Packages.tensorflow_2): log was empty</li>
<li>python38Packages.tflearn: log was empty</li>
  </ul>
</details>

**Comment by lukegb** [comment]
Hmm, I'm unconvinced by those build failures:

@ofborg build bazel-remote python37Packages.edward python37Packages.tensorflow python37Packages.tflearn python38Packages.edward python38Packages.tensorflow python38Packages.tflearn

**Comment by SuperSandro2000** [comment]
> Hmm, I'm unconvinced by those build failures:

Bazels is known to not work on darwin.

**Comment by lukegb** [comment]
Result of `nixpkgs-review pr 108741` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>26 packages marked as broken and skipped:</summary>
  <ul>
    <li>python37Packages.arviz</li>
    <li>python37Packages.dm-sonnet</li>
    <li>python37Packages.graph_nets</li>
    <li>python37Packages.optuna</li>
    <li>python37Packages.pymc3</li>
    <li>python37Packages.rl-coach</li>
    <li>python37Packages.tensorflow-probability</li>
    <li>python38Packages.arviz</li>
    <li>python38Packages.dm-sonnet</li>
    <li>python38Packages.graph_nets</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.pymc3</li>
    <li>python38Packages.rl-coach</li>
    <li>python38Packages.tensorflow-probability</li>
    <li>python39Packages.baselines</li>
    <li>python39Packages.dm-sonnet</li>
    <li>python39Packages.edward</li>
    <li>python39Packages.graph_nets</li>
    <li>python39Packages.tensorflow</li>
    <li>python39Packages.tensorflow-build</li>
    <li>python39Packages.tensorflow-build_2</li>
    <li>python39Packages.tensorflow-probability</li>
    <li>python39Packages.tensorflowWithCuda</li>
    <li>python39Packages.tensorflowWithoutCuda</li>
    <li>python39Packages.tensorflow_2</li>
    <li>python39Packages.tflearn</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>tts</li>
  </ul>
</details>
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>bazel-remote</li>
    <li>bazel-watcher</li>
    <li>envoy</li>
    <li>gvisor</li>
    <li>python37Packages.baselines</li>
    <li>python37Packages.edward</li>
    <li>python37Packages.mask-rcnn</li>
    <li>python37Packages.tensorflow (python37Packages.tensorflow-build ,python37Packages.tensorflow-build_2 ,python37Packages.tensorflowWithoutCuda ,python37Packages.tensorflow_2)</li>
    <li>python37Packages.tensorflowWithCuda</li>
    <li>python37Packages.tflearn</li>
    <li>python38Packages.baselines</li>
    <li>python38Packages.edward</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.tensorflow (python38Packages.tensorflow-build ,python38Packages.tensorflow-build_2 ,python38Packages.tensorflowWithoutCuda ,python38Packages.tensorflow_2)</li>
    <li>python38Packages.tensorflowWithCuda</li>
    <li>python38Packages.tflearn</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
tts is broke on master.

**Review by SuperSandro2000 — APPROVED** [review]
should be good

**Pull request merged**